spec/models/user_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "spec_helper"
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+
+  describe "#association" do
+    it {should respond_to(:documents)}
+    it {should respond_to(:comments)}
+    it {should respond_to(:favorites)}
+    it {should respond_to(:histories)}
+    it {should respond_to(:fav_docs)}
+    it {should respond_to(:categories)}
+  end
+
+  describe "#password" do
+    it {should have_secure_password}
+  end
+
+  describe "#scopes" do
+    context "sort by name ascending" do
+      it "should be valid" do
+        expect(User.sort_by_name.to_sql).to eq User.all.order(:fullname).to_sql
+      end
+    end
+
+    context "find in month" do
+      it "should be valid" do
+        expect(User.find_in_month.to_sql).to eq User.all.where("created_at BETWEEN ? AND ?", DateTime.now.beginning_of_month, DateTime.now.end_of_month).to_sql
+      end
+    end
+  end
+
+  describe "#validations" do
+    let!(:user) {FactoryBot.create :user}
```

Comment:
` let!` chị dùng `let` thôi dc ko, 
let và let! thế nào nhé

Reply:
dạ let thì trong mỗi block it thì nó sẽ gọi lại code trong lambda nên giá trị có thể khác nhau, còn let! thì giá trị sẽ được gán trước khi thực hiện và giá trị ở mỗi block it sẽ là như nhau. Ở đây em chỉ cần 1 biến user để test tất cả các block it trong describe validation nên dùng let! sẽ hợp lý hơn.